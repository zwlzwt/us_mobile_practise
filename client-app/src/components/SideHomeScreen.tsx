import React from 'react';
import styled from 'styled-components';
import backgroundImg from '../images/landing-hero-mobile.png';
import twitter from '../images/Twitter.png';
import instagram from '../images/Instagram.png';
import linkedin from '../images/Linkedin.png';
import facebook from '../images/Facebook.png';


const Background = styled.aside`
  display: inline-block;
  width: 427.14px;
  height: 100vh;
  background: no-repeat center url(${backgroundImg});
  background-size: 100% 100%;

  @media screen and (max-width: 414px) {
    display: none;
  }
`

const Title = styled.div`
  display: flex;
  margin-top: 86px;
  color: #fff;
  flex-direction: column;
  justify-content: center;
  z-index: 2;
  align-items: center;
  font-style: normal;
  font-weight: bold;
  font-size: 26px;
  line-height: 29px;
  letter-spacing: -1.2px;
`

const AboutUs = styled.div`
  position: absolute;
  display: flex;
  justify-content: space-around;
  align-items: center;
  bottom: 42px;
  width: 427.14px;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 16px;

  letter-spacing: 0.6px;
  text-transform: uppercase;

  color: #FFFFFF;

  img {
    margin-left: 12px;
  }
`


const SideHomeScreen = () => {
  return (
    <Background>
      <Title>
        <span>A few clicks away from your</span>
        <span>US Mobile account</span>
      </Title>

      <AboutUs>
        <p>About us Mobile</p>
        <div>
          <img src={twitter} alt='twitter'/>
          <img src={linkedin} alt='linkedin'/>
          <img src={facebook} alt='facebook'/>
          <img src={instagram} alt='instagram'/>
        </div>
      </AboutUs>
    </Background>
  )
}

export default SideHomeScreen