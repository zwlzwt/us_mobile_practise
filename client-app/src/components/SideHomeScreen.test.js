import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import SideHomeScreen from './SideHomeScreen';


it('renders correctly react-test-renderer', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<SideHomeScreen />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});