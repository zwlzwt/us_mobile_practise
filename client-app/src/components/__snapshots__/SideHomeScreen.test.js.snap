// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly react-test-renderer 1`] = `
<ForwardRef(styled.aside)>
  <ForwardRef(styled.div)>
    <span>
      A few clicks away from your
    </span>
    <span>
      US Mobile account
    </span>
  </ForwardRef(styled.div)>
  <ForwardRef(styled.div)>
    <p>
      About us Mobile
    </p>
    <div>
      <img
        alt="twitter"
        src="Twitter.png"
      />
      <img
        alt="linkedin"
        src="Linkedin.png"
      />
      <img
        alt="facebook"
        src="Facebook.png"
      />
      <img
        alt="instagram"
        src="Instagram.png"
      />
    </div>
  </ForwardRef(styled.div)>
</ForwardRef(styled.aside)>
`;
